telegram/_ownedgift.py

```diff
@@ -0,0 +1,415 @@
+            cls.REGULAR: OwnedGiftRegular,
+            cls.UNIQUE: OwnedGiftUnique,
+        }
+
+        if cls is OwnedGift and data.get("type") in _class_mapping:
+            return _class_mapping[data.pop("type")].de_json(data=data, bot=bot)
+
+        return super().de_json(data=data, bot=bot)
+
+
+class OwnedGifts(TelegramObject):
+    """Contains the list of gifts received and owned by a user or a chat.
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`total_count` and :attr:`gifts` are equal.
+
+    .. versionadded:: NEXT.VERSION
+
+    Args:
+        total_count (:obj:`int`): The total number of gifts owned by the user or the chat.
+        gifts (Sequence[:class:`telegram.OwnedGift`]): The list of gifts.
+        next_offset (:obj:`str`, optional): Offset for the next request. If empty,
+            then there are no more results.
+
+    Attributes:
+        total_count (:obj:`int`): The total number of gifts owned by the user or the chat.
+        gifts (Sequence[:class:`telegram.OwnedGift`]): The list of gifts.
+        next_offset (:obj:`str`): Optional. Offset for the next request. If empty,
+            then there are no more results.
+    """
+
+    __slots__ = (
+        "gifts",
+        "next_offset",
+        "total_count",
+    )
+
+    def __init__(
+        self,
+        total_count: int,
+        gifts: Sequence[OwnedGift],
+        next_offset: Optional[str] = None,
+        *,
+        api_kwargs: Optional[JSONDict] = None,
+    ):
+        super().__init__(api_kwargs=api_kwargs)
+        self.total_count: int = total_count
+        self.gifts: tuple[OwnedGift, ...] = parse_sequence_arg(gifts)
+        self.next_offset: Optional[str] = next_offset
+
+        self._id_attrs = (self.total_count, self.gifts)
+
+        self._freeze()
+
+    @classmethod
+    def de_json(cls, data: JSONDict, bot: Optional["Bot"] = None) -> "OwnedGifts":
+        """See :meth:`telegram.TelegramObject.de_json`."""
+        data = cls._parse_data(data)
+
+        data["gifts"] = de_list_optional(data.get("gifts"), OwnedGift, bot)
+        return super().de_json(data=data, bot=bot)
+
+
+class OwnedGiftRegular(OwnedGift):
+    """Describes a regular gift owned by a user or a chat.
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`gift` and :attr:`send_date` are equal.
+
+    .. versionadded:: NEXT.VERSION
+
+    Args:
+        gift (:class:`telegram.Gift`): Information about the regular gift.
+        owned_gift_id (:obj:`str`, optional): Unique identifier of the gift for the bot; for
+            gifts received on behalf of business accounts only.
+        sender_user (:class:`telegram.User`, optional): Sender of the gift if it is a known user.
+        send_date (:obj:`datetime.datetime`): Date the gift was sent in Unix time.
```

Comment:
```suggestion
        send_date (:obj:`datetime.datetime`): Date the gift was sent as :class:`datetime.datetime`. |datetime_localization|
```
same for the attribute :)